examples/async-recursion/src/lib.rs

```diff
@@ -0,0 +1,44 @@
+#![no_std]
+
+use core::num::ParseIntError;
+use gstd::{msg, prelude::*, ActorId};
+use async_recursion::async_recursion;
+
+static mut DEST: ActorId = ActorId::new([0u8; 32]);
+
+fn decode_hex(s: &str) -> Result<Vec<u8>, ParseIntError> {
+    (0..s.len())
+        .step_by(2)
+        .map(|i| u8::from_str_radix(&s[i..i + 2], 16))
+        .collect()
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn init() {
+    let input = String::from_utf8(msg::load_bytes()).expect("Invalid message: should be utf-8");
+    DEST = ActorId::from_slice(
+        &decode_hex(&input).expect("INTIALIZATION FAILED: INVALID PROGRAM ID"),
+    )
+    .expect("Unable to create ActorId");
```

Comment:
ActorId supports codec so the code can be more type-safe but `decode_hex` can be removed (which is already copypasted around):
```suggestion
gstd::metadata! {
    title: "demo async recursion",
    init:
        input: ActorId,
}

#[no_mangle]
pub unsafe extern "C" fn init() {
    DEST = msg::load().expect("Failed to decode `ActorId`");
```

Reply:
Unfortunately this  solution is not working for our test system, because rtest and gtest do not encode ActorId, they just send it as utf-8.

Reply:
`kind: bytes` may be used with provided hex-encoded byte-string '0xABCDEF...789'